cpp/src/slice2cpp/Gen.cpp

```diff
@@ -898,289 +874,190 @@ Slice::Gen::writeExtraHeaders(IceInternal::Output& out)
 void
 Slice::Gen::validateMetadata(const UnitPtr& u)
 {
-    MetadataVisitor visitor;
-    u->visit(&visitor);
-}
-
-bool
-Slice::Gen::MetadataVisitor::visitUnitStart(const UnitPtr& unit)
-{
-    // Validate file metadata in the top-level file and all included files.
-    for (const string& file : unit->allFiles())
-    {
-        DefinitionContextPtr dc = unit->findDefinitionContext(file);
-        MetadataList fileMetadata = dc->getMetadata();
-        assert(dc);
-        bool seenHeaderExtension = false;
-        bool seenSourceExtension = false;
-        bool seenDllExport = false;
+    map<string, MetadataInfo> metadataInfo;
```

Comment:
The name of the variable is poorly chosen. A variable name should match a type name only when the variable is of that type (with possibly qualification/pointer and the like).

Here it should be something more like `validationMap`.